packages/config/src/VitessceAutoConfig.js

```diff
@@ -185,55 +185,91 @@ class AnndataZarrAutoConfig extends AbstractAutoConfig {
     return views;
   }
 
-  async setMetadataSummary() {
-    if (Object.keys(this.metadataSummary).length > 0) {
-      return this.metadataSummary;
+  async setMetadataSummaryWithZmetadata(response) { /* eslint-disable-line class-methods-use-this */
+    const metadataFile = await response.json();
+    if (!metadataFile.metadata) {
+      throw new Error('Could not generate config: .zmetadata file is not valid.');
     }
 
-    const parseMetadataFile = (metadataFile) => {
-      if (!metadataFile.metadata) {
-        throw new Error('Could not generate config: .zmetadata file is not valid.');
-      }
+    const obsmKeys = Object.keys(metadataFile.metadata)
+      .filter(key => key.startsWith('obsm/X_'))
+      .map(key => key.split('/.zarray')[0]);
 
-      const obsmKeys = Object.keys(metadataFile.metadata)
-        .filter(key => key.startsWith('obsm/X_'))
-        .map(key => key.split('/.zarray')[0]);
+    const obsKeysArr = Object.keys(metadataFile.metadata)
+      .filter(key => key.startsWith('obs/')).map(key => key.split('/.za')[0]);
 
-      const obsKeysArr = Object.keys(metadataFile.metadata)
-        .filter(key => key.startsWith('obs/')).map(key => key.split('/.za')[0]);
+    function uniq(a) {
+      return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
+    }
+    const obsKeys = uniq(obsKeysArr);
 
-      function uniq(a) {
-        return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
-      }
-      const obsKeys = uniq(obsKeysArr);
+    const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+
+    return {
+      obsm: obsmKeys,
+      obs: obsKeys,
+      X: X.length > 0,
+    };
+  }
 
-      const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+  async setMetadataSummaryWithoutZmetadata() {
+    const suffixes = [
+      '/obsm/X_pca/.zarray',
+      '/obsm/X_umap/.zarray',
+      '/obsm/X_tsne/.zarray',
+      '/obsm/X_spatial/.zarray',
+      '/obsm/X_segmentations/.zarray',
+      '/obs/.zattrs',
+      '/X/.zarray',
+      '/X/data/.zarray',
+    ];
 
-      const out = {
-        obsm: obsmKeys,
-        obs: obsKeys,
-        X: X.length > 0,
-      };
+    const promises = suffixes.map(suffix => fetch(this.fileUrl + suffix));
 
-      return out;
+    const results = await Promise.all(promises);
+    const metadataSummary = {
+      obsm: [],
+      obs: [],
+      X: false,
     };
 
+    const obsJson = results.find(j => j.ok && j.url.startsWith(`${this.fileUrl}/obs/.zattrs`));
+
+    if (obsJson) {
```

Comment:
in case the ways dataframes are stored changes in a future anndata version, we should either:
-  add extra checks that `'encoding-type': 'dataframe',
and  'encoding-version' is either '0.1.0'  or '0.2.0' and fail otherwise, or

- double check that `'column-order'` exists, is an array, and each value is a string

References:
https://anndata.readthedocs.io/en/0.7.8/fileformat-prose.html#dataframes
https://anndata.readthedocs.io/en/latest/fileformat-prose.html#dataframes

Reply:
Thanks for the tip! I have added a validation function before I access `column-order`.